### PR TITLE
Update documentation for Job deprecation

### PR DESCRIPTION
**Pull request by uellue**
* Update documentation and docstrings
* Move Task from Job to UDF code base
* Fix RawFFT that was broken from making UDF the default back-end
* Improve RawFFT tests to catch such issues in the future
* Drive-by change: Dask to >=2.2.0 and fix of parameter name

## Contributor Checklist:

* [x] I have added myself to [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes
* [x] I have added/updated test cases

### CODE REVIEW

File: src/libertem/job/base.py [hunk 1]
```diff
@@ -25,9 +36,19 @@ class BaseJob(Job):
...
+
     """
 
     def __init__(self, dataset):
+        log.warning(
+            "The Job API is deprecated and will be removed after version 0.6.0. See "
+            "https://libertem.github.io/LiberTEM/changelog.html#job-deprecation "
+            "for details and a migration guide. "
+            "Info: Instantiating class %s", type(self)
+        )
```
[the remark this answers is not in the document]
Comment (sk1p) [reply src/libertem/job/base.py]: Also, to give better guidance, the `create_pick_job` and `create_mask_job` methods should raise a `DeprecationWarning` that directly references the replacement, i.e. the matching `*_analysis` methods. AFAIK these warnings should also appear in ipython / jupyter notebooks.

If we want to go a step further (in the future?), we could have subclasses that encode the version when the feature will be removed, for example `RemovedIn06Warning`, [like Django's depreacation policy](https://docs.djangoproject.com/en/3.0/internals/release-process/#deprecation-policy). That way, for future-proofing, one can install a filter that errors on all  `RemovedIn05Warning`, while keeping `*06Warning`s as warnings.

File: test_requirements.txt [hunk 2]
```diff
@@ -3,3 +3,4 @@ pytest-cov
 pytest-asyncio
 aiohttp
 websockets
+testfixtures
```
[the remark this answers is not in the document]
Comment (uellue) [reply test_requirements.txt]: Nice catch! Yes, I forgot to git add

File: src/libertem/job/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (uellue) [reply src/libertem/job/base.py]: To show deprecation warnings to users in IPython, we can use a `FutureWarning`. The Job base class now throws a `DeprecationWarning` targeted at application developers, and the API throws a `FutureWarning` targeted at script developers, i.e. end users of LiberTEM. That message appears in Jupyter as it should. Additional benefit of only showing one message and not two! Sounds good? :-)

**Comment by sk1p** [comment]
CI failure is because of intersphinx failing to fetch numpy inventory file. Docs should be fine, as `docs-check` ran successfully.

File: src/libertem/job/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sk1p) [reply src/libertem/job/base.py]: > Sounds good? :-)

I think that's the best we can do for now - I guess one can argue that interactive / notebook usage can count as "end user" usage :grin:

**Comment by sk1p** [comment]
Looks good, thanks!

**Pull request merged**